logOutputSubject_test.go

```diff
@@ -91,6 +98,47 @@ func TestLogOutputSubject_OutputShouldCallFormatterAndWriter(t *testing.T) {
 	assert.Equal(t, int32(1), atomic.LoadInt32(&formatterCalled))
 }
 
+func TestLogOutputSubject_OutputShouldProduceCorrectString(t *testing.T) {
+	t.Parallel()
+
+	los := logger.NewLogOutputSubject()
+	var writtenData []byte
+	_ = los.AddObserver(
+		&mock.WriterStub{
+			WriteCalled: func(p []byte) (n int, err error) {
+				writtenData = p
+				return 0, nil
+			},
+		},
+		&mock.FormatterMock{},
+	)
+
+	logLine := &logger.LogLine{
+		LoggerName:  "",
+		Correlation: proto.LogCorrelationMessage{},
+		Message:     "message",
+		LogLevel:    logger.LogDebug,
+		Args: []interface{}{
+			"int", 1,
+			"ASCII string", "plain text \n",
+			"non-ASCII string", "Âµs",
+			"time.Duration", time.Microsecond*4 + time.Nanosecond,
+			"byte slice", []byte("aaa"),
+			"error", errors.New("an error"),
+			"bool", true,
+		},
+		Timestamp: time.Date(2022, 03, 30, 15, 47, 52, 0, time.Local),
+	}
+
+	los.Output(logLine)
+
+	expectedString := `DEBUG[2022-03-30 15:47:52.000]   message                                  int = 1 ASCII string = plain text 
+ non-ASCII string = c382c2b573 time.Duration = 4.001µs byte slice = 616161 error = an error bool = true 
```

Comment:
[question]

Do we want here to output the whole string as hex if only a part of it is non-ASCII?

E.g.: from `Âµs` -> `c382c2b5*s*` instead of `c382c2b573`

Reply:
full hex representation is desirable